configure.ac

```
@@ -543,6 +554,10 @@ case "${host}" in
 							   have_notify="no"
 							   have_gio2="no"  ])
 			LIBS="$saved_LIBS"
+			if test "${have_gio2}" = "yes"; then
+				# we do not need lglib-2.0
+				GIO2_LIBS="-lgio-2.0 -lgobject-2.0"
```

This breaks FreeBSD and other systems with `/usr/local`:

```
> /usr/local/bin/pkg-config --libs gio-2.0
-L/usr/local/lib -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl
```

Instead of this, plain `-lgio-2.0 -lgobject-2.0 ` is used and it causes the following build failure:

```
Making all in libopensc
/bin/sh ../../libtool  --tag CC   --mode=link cc     -I/usr/local/include/PCSC -pthread   -g -O2   -version-info 9:0:1  -export-symbols "../../../OpenSC/src/libopensc/libopensc.exports"  -no-undefined  -o libopensc.la -rpath /usr/local/lib libopensc_la-sc.lo libopensc_la-ctx.lo  libopensc_la-log.lo libopensc_la-errors.lo  libopensc_la-asn1.lo libopensc_la-base64.lo  libopensc_la-sec.lo libopensc_la-card.lo  libopensc_la-iso7816.lo libopensc_la-dir.lo  libopensc_la-ef-atr.lo libopensc_la-ef-gdo.lo  libopensc_la-padding.lo libopensc_la-apdu.lo  libopensc_la-simpletlv.lo libopensc_la-gp.lo  libopensc_la-pkcs15.lo libopensc_la-pkcs15-cert.lo  libopensc_la-pkcs15-data.lo libopensc_la-pkcs15-pin.lo  libopensc_la-pkcs15-prkey.lo libopensc_la-pkcs15-pubkey.lo  libopensc_la-pkcs15-skey.lo libopensc_la-pkcs15-sec.lo  libopensc_la-pkcs15-algo.lo libopensc_la-pkcs15-cache.lo  libopensc_la-pkcs15-syn.lo  libopensc_la-pkcs15-emulator-filter.lo libopensc_la-muscle.lo  libopensc_la-muscle-filesystem.lo libopensc_la-ctbcs.lo  libopensc_la-reader-ctapi.lo libopensc_la-reader-pcsc.lo  libopensc_la-reader-openct.lo libopensc_la-reader-tr03119.lo  libopensc_la-card-setcos.lo libopensc_la-card-flex.lo  libopensc_la-card-gpk.lo libopensc_la-card-cardos.lo  libopensc_la-card-tcos.lo libopensc_la-card-default.lo  libopensc_la-card-mcrd.lo libopensc_la-card-starcos.lo  libopensc_la-card-openpgp.lo libopensc_la-card-oberthur.lo  libopensc_la-card-belpic.lo libopensc_la-card-atrust-acos.lo  libopensc_la-card-entersafe.lo libopensc_la-card-epass2003.lo  libopensc_la-card-coolkey.lo libopensc_la-card-incrypto34.lo  libopensc_la-card-piv.lo libopensc_la-card-cac-common.lo  libopensc_la-card-cac.lo libopensc_la-card-cac1.lo  libopensc_la-card-muscle.lo libopensc_la-card-asepcos.lo  libopensc_la-card-akis.lo libopensc_la-card-gemsafeV1.lo  libopensc_la-card-rutoken.lo libopensc_la-card-rtecp.lo  libopensc_la-card-westcos.lo libopensc_la-card-myeid.lo  libopensc_la-card-itacns.lo libopensc_la-card-authentic.lo  libopensc_la-card-iasecc.lo libopensc_la-iasecc-sdo.lo  libopensc_la-iasecc-sm.lo libopensc_la-card-sc-hsm.lo  libopensc_la-card-dnie.lo libopensc_la-cwa14890.lo  libopensc_la-cwa-dnie.lo libopensc_la-card-isoApplet.lo  libopensc_la-card-masktech.lo libopensc_la-card-gids.lo  libopensc_la-card-jpki.lo libopensc_la-card-npa.lo  libopensc_la-card-esteid2018.lo libopensc_la-card-idprime.lo  libopensc_la-card-edo.lo libopensc_la-card-nqApplet.lo  libopensc_la-pkcs15-openpgp.lo libopensc_la-pkcs15-starcert.lo  libopensc_la-pkcs15-cardos.lo libopensc_la-pkcs15-tcos.lo  libopensc_la-pkcs15-esteid.lo  libopensc_la-pkcs15-gemsafeGPK.lo  libopensc_la-pkcs15-actalis.lo  libopensc_la-pkcs15-atrust-acos.lo  libopensc_la-pkcs15-tccardos.lo libopensc_la-pkcs15-piv.lo  libopensc_la-pkcs15-cac.lo libopensc_la-pkcs15-esinit.lo  libopensc_la-pkcs15-westcos.lo libopensc_la-pkcs15-pteid.lo  libopensc_la-pkcs15-oberthur.lo libopensc_la-pkcs15-itacns.lo  libopensc_la-pkcs15-gemsafeV1.lo libopensc_la-pkcs15-sc-hsm.lo  libopensc_la-pkcs15-coolkey.lo  libopensc_la-pkcs15-din-66291.lo  libopensc_la-pkcs15-idprime.lo libopensc_la-pkcs15-nqApplet.lo  libopensc_la-pkcs15-dnie.lo libopensc_la-pkcs15-gids.lo  libopensc_la-pkcs15-iasecc.lo libopensc_la-pkcs15-jpki.lo  libopensc_la-pkcs15-esteid2018.lo  libopensc_la-pkcs15-starcos-esign.lo  libopensc_la-compression.lo libopensc_la-sm.lo  libopensc_la-aux-data.lo    -lcrypto   -lz  ../../src/pkcs15init/libpkcs15init.la  ../../src/scconf/libscconf.la  ../../src/common/libscdl.la  ../../src/ui/libnotify.la  ../../src/ui/libstrings.la  ../../src/sm/libsmeac.la  ../../src/common/libcompat.la  
libtool: link: cc -shared  -fPIC -DPIC  .libs/libopensc_la-sc.o .libs/libopensc_la-ctx.o .libs/libopensc_la-log.o .libs/libopensc_la-errors.o .libs/libopensc_la-asn1.o .libs/libopensc_la-base64.o .libs/libopensc_la-sec.o .libs/libopensc_la-card.o .libs/libopensc_la-iso7816.o .libs/libopensc_la-dir.o .libs/libopensc_la-ef-atr.o .libs/libopensc_la-ef-gdo.o .libs/libopensc_la-padding.o .libs/libopensc_la-apdu.o .libs/libopensc_la-simpletlv.o .libs/libopensc_la-gp.o .libs/libopensc_la-pkcs15.o .libs/libopensc_la-pkcs15-cert.o .libs/libopensc_la-pkcs15-data.o .libs/libopensc_la-pkcs15-pin.o .libs/libopensc_la-pkcs15-prkey.o .libs/libopensc_la-pkcs15-pubkey.o .libs/libopensc_la-pkcs15-skey.o .libs/libopensc_la-pkcs15-sec.o .libs/libopensc_la-pkcs15-algo.o .libs/libopensc_la-pkcs15-cache.o .libs/libopensc_la-pkcs15-syn.o .libs/libopensc_la-pkcs15-emulator-filter.o .libs/libopensc_la-muscle.o .libs/libopensc_la-muscle-filesystem.o .libs/libopensc_la-ctbcs.o .libs/libopensc_la-reader-ctapi.o .libs/libopensc_la-reader-pcsc.o .libs/libopensc_la-reader-openct.o .libs/libopensc_la-reader-tr03119.o .libs/libopensc_la-card-setcos.o .libs/libopensc_la-card-flex.o .libs/libopensc_la-card-gpk.o .libs/libopensc_la-card-cardos.o .libs/libopensc_la-card-tcos.o .libs/libopensc_la-card-default.o .libs/libopensc_la-card-mcrd.o .libs/libopensc_la-card-starcos.o .libs/libopensc_la-card-openpgp.o .libs/libopensc_la-card-oberthur.o .libs/libopensc_la-card-belpic.o .libs/libopensc_la-card-atrust-acos.o .libs/libopensc_la-card-entersafe.o .libs/libopensc_la-card-epass2003.o .libs/libopensc_la-card-coolkey.o .libs/libopensc_la-card-incrypto34.o .libs/libopensc_la-card-piv.o .libs/libopensc_la-card-cac-common.o .libs/libopensc_la-card-cac.o .libs/libopensc_la-card-cac1.o .libs/libopensc_la-card-muscle.o .libs/libopensc_la-card-asepcos.o .libs/libopensc_la-card-akis.o .libs/libopensc_la-card-gemsafeV1.o .libs/libopensc_la-card-rutoken.o .libs/libopensc_la-card-rtecp.o .libs/libopensc_la-card-westcos.o .libs/libopensc_la-card-myeid.o .libs/libopensc_la-card-itacns.o .libs/libopensc_la-card-authentic.o .libs/libopensc_la-card-iasecc.o .libs/libopensc_la-iasecc-sdo.o .libs/libopensc_la-iasecc-sm.o .libs/libopensc_la-card-sc-hsm.o .libs/libopensc_la-card-dnie.o .libs/libopensc_la-cwa14890.o .libs/libopensc_la-cwa-dnie.o .libs/libopensc_la-card-isoApplet.o .libs/libopensc_la-card-masktech.o .libs/libopensc_la-card-gids.o .libs/libopensc_la-card-jpki.o .libs/libopensc_la-card-npa.o .libs/libopensc_la-card-esteid2018.o .libs/libopensc_la-card-idprime.o .libs/libopensc_la-card-edo.o .libs/libopensc_la-card-nqApplet.o .libs/libopensc_la-pkcs15-openpgp.o .libs/libopensc_la-pkcs15-starcert.o .libs/libopensc_la-pkcs15-cardos.o .libs/libopensc_la-pkcs15-tcos.o .libs/libopensc_la-pkcs15-esteid.o .libs/libopensc_la-pkcs15-gemsafeGPK.o .libs/libopensc_la-pkcs15-actalis.o .libs/libopensc_la-pkcs15-atrust-acos.o .libs/libopensc_la-pkcs15-tccardos.o .libs/libopensc_la-pkcs15-piv.o .libs/libopensc_la-pkcs15-cac.o .libs/libopensc_la-pkcs15-esinit.o .libs/libopensc_la-pkcs15-westcos.o .libs/libopensc_la-pkcs15-pteid.o .libs/libopensc_la-pkcs15-oberthur.o .libs/libopensc_la-pkcs15-itacns.o .libs/libopensc_la-pkcs15-gemsafeV1.o .libs/libopensc_la-pkcs15-sc-hsm.o .libs/libopensc_la-pkcs15-coolkey.o .libs/libopensc_la-pkcs15-din-66291.o .libs/libopensc_la-pkcs15-idprime.o .libs/libopensc_la-pkcs15-nqApplet.o .libs/libopensc_la-pkcs15-dnie.o .libs/libopensc_la-pkcs15-gids.o .libs/libopensc_la-pkcs15-iasecc.o .libs/libopensc_la-pkcs15-jpki.o .libs/libopensc_la-pkcs15-esteid2018.o .libs/libopensc_la-pkcs15-starcos-esign.o .libs/libopensc_la-compression.o .libs/libopensc_la-sm.o .libs/libopensc_la-aux-data.o  -Wl,--whole-archive ../../src/pkcs15init/.libs/libpkcs15init.a ../../src/scconf/.libs/libscconf.a ../../src/common/.libs/libscdl.a ../../src/ui/.libs/libnotify.a ../../src/ui/.libs/libstrings.a ../../src/sm/.libs/libsmeac.a ../../src/common/.libs/libcompat.a -Wl,--no-whole-archive  -lz -ldl -lgio-2.0 -lgobject-2.0 -lcrypto  -g -O2   -pthread -Wl,-soname -Wl,libopensc.so.8 -Wl,-retain-symbols-file -Wl,../../../OpenSC/src/libopensc/libopensc.exports -o .libs/libopensc.so.8.1.0
ld: error: unable to find library -lgio-2.0
ld: error: unable to find library -lgobject-2.0
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** Error code 1
```

This is because the `-L` flag as given by `pkg-config` is missing.

---

Indeed, fiddeling with the libraries and paths can be difficult.

@xhanulik would it be possible to use the pkg-config output and pass it through `sed` to just remove `-lglib-2.0`?

---

yes, pkg-config sets the `GIO2_LIBS` variable, so something like
```suggestion
				GIO2_LIBS=$(echo "$GIO2_LIBS" | sed -e 's/-lglib-2.0//g')
```
would remove only `-lglib-2.0`